catroid/src/org/catrobat/catroid/content/actions/ChangeColorByNAction.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2016 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.content.actions;
+
+import android.util.Log;
+
+import com.badlogic.gdx.scenes.scene2d.actions.TemporalAction;
+
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.formulaeditor.Formula;
+import org.catrobat.catroid.formulaeditor.InterpretationException;
+
+public class ChangeColorByNAction extends TemporalAction {
+
+	private Sprite sprite;
+	private Formula color;
+
+	protected void update(float delta) {
+		try {
+			float eightBitColor = color == null ? 25 : color.interpretFloat(sprite);
+			sprite.look.changeColorInUserInterfaceDimensionUnit(eightBitColor);
+		} catch (InterpretationException interpretationException) {
+			Log.d(getClass().getSimpleName(), "Formula interpretation for this specific Brick failed.", interpretationException);
```

Comment:
This line was copied from one of the many other actions. :wink: If the color formula doesn't evaluate to a number, the action is ignored. It's the expected behavior for this action and others like SetX, GoNStepsBackAction and so on.

So `Log.d` is fine. `Log.e` should only be used when something _fatal_ has happened and _won't be recoverable_.

https://source.android.com/source/code-style.html#log-sparingly, not our logging guidelines but still useful.
You may argue that it should not be logged at all, be logged as verbose instead or logging a stack trace is unnecessary. However, for consistency, I suggest to keep it the way it is. Feel free to create a Jira ticket which addresses _every_ action if you want to change something.